config/src/main/java/com/typesafe/config/impl/AbstractConfigValue.java

```diff
@@ -27,10 +27,21 @@
 
     final private SimpleConfigOrigin origin;
 
+    private boolean optional;
```

Comment:
this field won't work for a couple reasons; 1) ConfigValue is immutable, 2) if this was changed to make a copy of the config value, it would still not be conceptually part of the value, it's part of the implied "schema" instead.

will suggest another way to do this below